File: ts/1.array/non-constructible-change.ts
export function nonConstructibleChange(arr: number[]) {
    arr.sort((a, b) => {
        return a > b ? 1 : -1;
    });

    let change = 0;
    for (let i = 0; i < arr.length; ++i) {
        let num = arr[i];
        if (change + 1 < num) {
            return change + 1;
        }
        change += num;
    }

    return change + 1;
}

File: golang/1.array/non_constructible_change.go
package main

import "sort"

func findNonConstructibleChange(arr []int) int {
	sort.Ints(arr)
	change := 0
	nonConstructibleChange := false

	for _, value := range arr {
		if change+1 < value {
			nonConstructibleChange = true
			break
		}

		change += value
	}

	if nonConstructibleChange {
		change = change + 1
	}

	return change
}
